Sources/Workspace/Workspace+BinaryArtifacts.swift

```diff
@@ -455,30 +875,37 @@ extension Workspace.BinaryArtifactsManager {
             let fileName: String
             let checksum: String
             let supportedTriples: [Triple]
-
-            enum CodingKeys: String, CodingKey {
-                case fileName
-                case checksum
-                case supportedTriples
-            }
-
-            public init(from decoder: Decoder) throws {
-                let container = try decoder.container(keyedBy: CodingKeys.self)
-                self.fileName = try container.decode(String.self, forKey: .fileName)
-                self.checksum = try container.decode(String.self, forKey: .checksum)
-                self.supportedTriples = try container.decode([String].self, forKey: .supportedTriples).map(Triple.init)
-            }
         }
     }
 }
 
 extension Workspace.BinaryArtifactsManager {
+    struct ArchiveWrapperFile: Decodable {
+        let schemaVersion: String
+        let url: URL
+        let checksum: String
+        let path: RelativePath
+        let supportedTriples: [Triple]
+    }
+}
```

Comment:
the new wrapper file format will be a JSON containing these fields

Reply:
may need to make it more complicated with array of variants so you can have path -> supported triple all in one zip

Reply:
If I understand this correctly we would have a zip file that just contains a JSON file. That JSON file contains the following

```
{
    "schemaVersion": "1.0",
     "variants": [
           {
                “url”: “<url/to/artifact.zip>”,
                 “checksum”: “<checksum of the zip>”,
                "supportedTriples": [ "<triple1>", ... ]
            },     ]
}
```

Is my assumption correct here?

Reply:
almost

```
{
    "schemaVersion": "1.0",
     "variants": [
           {
                “url”: “<url/to/artifact.zip>”,
                 “checksum”: “<checksum of the zip>”,
                "supportedTriples": [<triple>, <triple>],
                 "path": <realtive path inside the zip file where the tools are located>
            },     
        ]
}
```

the idea here is that this would work similar to the bundle index file. the initial request get a JSON file (not zip) with the format like ^^, then SwiftPM will find the best variant based on triple and download the zip file and set it up using the path